vms/platformvm/state/diff.go

```diff
@@ -65,9 +68,14 @@ func NewDiff(
 	if !ok {
 		return nil, fmt.Errorf("%w: %s", ErrMissingParentState, parentID)
 	}
+	cfg, err := parentState.GetConfig()
```

Comment:
I want tempted to pass config as `NewDiff` parameter, but this would allow freedom to create staked diffs with different configurations. I believe the new `state.Chain` method `GetConfig` helps avoiding this.